signature/signature.go

```diff
@@ -0,0 +1,142 @@
+package signature
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const tsHeader = "MessageBird-Request-Timestamp"
+const sHeader = "MessageBird-Signature"
+
+// ValidityPeriod is the time in hours after which a request is descarded
+type ValidityPeriod *float64
```

Comment:
As a suggestion:
1. It doesn't need to be a pointer as far as I can see(?)
2. `time.Duration` would work perfectly fine as well and would be more readable(?)

Reply:
Its a pointer so it can be set to nil when no time validation is needed

Reply:
If you would not use hours as a period units in favour of `time.Duration` then 0 (which is a default value for time.Duration) could mean that there's no time validation. I believe sometimes it doesn't make sense to create custom types especially when you have well known type in the stdlib

Reply:
But it's up to you